components/concept/menu/src/main/java/mozilla/components/concept/menu/candidate/MenuCandidate.kt

```diff
@@ -78,6 +78,30 @@ data class CompoundMenuCandidate(
     }
 }
 
+/**
+ * Menu option that opens a nested sub menu.
+ *
+ * @property id Unique ID for this nested menu. Can be a resource ID.
+ * @property text Text to display.
+ * @property start Icon to display before the text.
+ * @property end Icon to display after the text.
+ * @property subMenu Nested menu items to display.
+ * If null, this item will instead return to the root menu.
+ * @property textStyle Styling to apply to the text.
+ * @property containerStyle Styling to apply to the container.
+ * @property effect Effects to apply to the option.
+ */
+data class NestedMenuCandidate(
```

Comment:
I think the correct ADT naming here would be `ParentMenuCandidate`. But I think `nested` is fine here as well.

Reply:
I'd prefer nested here